### PR TITLE
refactor(Kconfig): Set USB default based on hardware capabilities

### PR DESCRIPTION
**Pull request by Nicell**
Removes the USB default under every split right half in favor of a default based on hardware capabilities in the main app Kconfig.

### CODE REVIEW

**Comment by Nicell** [comment]
This PR should watch for any [split shields that merge](https://github.com/zmkfirmware/zmk/pulls?q=is%3Aopen+is%3Apr+label%3Ashields).

File: app/Kconfig [hunk 1]
```diff
@@ -317,6 +317,9 @@ config KERNEL_BIN_NAME
 config REBOOT
 	default y
 
+config USB
+	default y if HAS_HW_NRF_USBD
```
Comment (petejohanson) [inline app/Kconfig:321]: For our stm32 target, and soon SAMD21 targets, can we also set `y` for `USB_DC_STM32` and `USB_DC_SAM`?

Comment (petejohanson) [reply app/Kconfig:321]: @Nicell Ok, can we ditch the SAMD stuff for now, and leave the STM32, until I can dig further? Sooner we merge this, the less conflicts.

**Review by petejohanson — APPROVED** [review]
Thanks!

**Pull request merged**